markdown/events/2021/bytesize-9-json-schema.md

```diff
@@ -34,3 +34,154 @@ This will cover:
 The talk will be presented on Zoom and live-streamed on YouTube:
 * YouTube: <https://youtu.be/PU6vAj_7WRM>
 * Slides: <https://doi.org/10.6084/m9.figshare.14453214.v1>
+
+<details markdown="1"><summary>Video transcription</summary>
+**Note: The content has been edited to make it reader-friendly**
+
+[0:36](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=36) Welcome everybody, today's topic is JSON schema, what it is, why we chose to use it, and how it can be utilised.
+
+[0:53](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=53) First we need to talk about parameters. In NextFlow you either have them on the command line with -- in front of it, for example in `main.nf` and in `params.foo`.
+
+[1:12](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=72) When we look through all the pipelines in nf-core, from what I could parse, I found that we have a large number of different parameters, 47 on average but it can go up to 470.
+
+To make this usable, it should be documented so you can read up both as a user and as a developer about what these parameters do.
+
+In the best case, we have a validation and a background so it already throws an error before the tool that uses the parameter. Therefore it is easier to debug.
+
+But as you can imagine, with so many parameters it’s not so easy anymore to manage.
+
+[2:19](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=134) So we therefore looked at JSON schema and thought that this is exactly what we want.
+
+It is a standardised way to describe a data object, and it comes with its own validation library, which takes a load off of our shoulders.
+
+[2:32](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=154) Last summer, we added the `nextflow_schema.json` file to the template, which is a description of all the pipeline parameters in a JSON schema format.
+
+[2:47](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=167) Here you see an excerpt of it, and this is the only thing you should actually of the JSON because we are of the opinion that you should never have to need to interact with these complex files.
```

Comment:
```suggestion
[2:47](https://youtu.be/PU6vAj_7WRM?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=167) Here you see an excerpt of it, and this is the only thing you should actually see of the JSON because we are of the opinion that you should never have to need to interact with these complex files.
```